news-catalyst/google-app-scripts
# Link a document to an existing article

👋  I'd love some feedback on this feature. I consider this version 0.0.0(.0?) but it's functional (hopefully!). This is admittedly a bit rough but I'd rather get your take on how it's working so far instead of waiting on feedback 🙌  Thanks!

This PR lets you take any Google Document and link it to an existing article in Webiny.

To test:

* Open a google doc not currently associated with a webiny article; or use [my test example](https://docs.google.com/document/d/1HjpiubyneR7pGQeW_BFQqpOwKADiFG7ra0NtF51ZS_c/edit?usp=sharing), which is associated (either way works)
* Test using the add-on at "latest code" or version 40: "Manually link doc->article v0"
* Note 2 submenu items for the add-on: Publishing Tools (sidebar we've been using) and Administrator Tools (new)
* Open Administrator tools
* search for an article by headline (try this: `manually`; it's a test article)
* currently only one locale is available, so it's selected
* click "Link"

The document should now be associated with the existing article.

What this means:

* google doc "properties" now stores the articleID
* subsequent loading of administrator tools sidebar should indicate the doc is already linked
* previous article content in webiny is REPLACED with current document contents
* you need to use "Publishing Tools" sidebar to edit the article metadata (including tags, category) or publish it (first time or updates)

I feel like there's still a bit to consider here, but off the cuff some to-do's:

* [x] flip `published` to `false` when saving new linked article in db
* [ ] will we need to limit access to this tool? if so, how? (e.g. to editors; to admins)
* [x] confirmation dialog step added to "<Link>" submit button
* [x] test with multiple locales
* [x] remove the "show/hide" configuration buttons from Publishing Tools sidebar; should live in Administrator Tools
* [x] move the "delete article" functionality from Publishing tools to Admin tools
* [x] freshly request the article data from webiny IF there's an article ID when the sidebar is loading
* [x] couple bugfixes on storage of contents & category
* all kinds of UI/UX thinking and implementation
